src/explore-education-statistics-frontend/src/prototypes/components/PrototypeSearchForm.tsx

```diff
@@ -125,9 +114,60 @@ class PrototypeSearchForm extends Component<{}, State> {
     this.setState({ searchResults });
   }
 
-  private handleTextChange = debounce(() => {
-    this.performSearch();
-  }, 1000);
+  private static openAccordion(potentialAccordion: HTMLElement) {
+    potentialAccordion.classList.add('govuk-accordion__section--expanded');
+  }
+
+  private resetSearch() {
+    this.setState({
+      currentlyHighlighted: undefined,
+      searchResults: [],
+      searchValue: '',
+    });
+  }
+
+  private static calculateLocationOfElement(
+    element: HTMLElement,
+    insideAccordion: boolean,
+    potentialAccordion: HTMLElement,
+  ) {
+    const location = [];
+
+    const locationHeaderElement = PrototypeSearchForm.findSiblingsBeforeOfElementType(
+      element,
+      'h3',
+      'h2',
+      'h4',
+    );
+
+    if (locationHeaderElement) {
+      location.unshift(
+        PrototypeSearchForm.substring(
+          locationHeaderElement.textContent || '',
+          20,
+        ),
+      );
+    }
+
+    if (insideAccordion) {
+      const accordionHeader = potentialAccordion.querySelector(
+        '.govuk-accordion__section-heading button',
+      );
+      if (accordionHeader) {
+        location.unshift(
+          PrototypeSearchForm.substring(accordionHeader.textContent || '', 20),
+        );
+      }
+    }
+
+    return location;
+  }
+
+  private handleTextChange() {
+    debounce(() => {
```

Comment:
Does this actually debounce properly? I would have thought you'd have to assign the debounced function into a class property

Reply:
I merged the debounce changes in from the master branch. I didn't have the debounce function in there are first.